flink-runtime/src/main/java/org/apache/flink/runtime/state/StreamStateHandle.java

```diff
@@ -38,6 +38,14 @@ public interface StreamStateHandle extends StateObject {
     /** @return Content of this handle as bytes array if it is already in memory. */
     Optional<byte[]> asBytesIfInMemory();
 
+    /**
+     * @return Path to an underlying file represented by this {@link StreamStateHandle} or {@link
+     *     Optional#empty()} if there is no such file.
+     */
+    default Optional<org.apache.flink.core.fs.Path> maybeGetPath() {
+        return Optional.empty();
+    }
```

Comment:
Should this also be implemented by `DirectoryStreamStateHandle` and `SegmentFileStateHandle`?

Reply:
Ok, I see.

> And this wouldn't necessarily mean that s5 will be used for these handles - it also depends on the caller (rocksdb downloader).

I somehow have a feeling that in the future we might want to rethink how to decide which downloading/uploading code path to use.